File: Ceres-Middleware-Location/src/main/java/cn/cerestech/middleware/location/entity/Address.java
package cn.cerestech.middleware.location.entity;

import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.Table;

import cn.cerestech.framework.support.persistence.entity.IdEntity;
import cn.cerestech.middleware.location.enums.AdminLevel;

@SuppressWarnings("serial")
@Entity
@Table(name = "$$sys_address")
public class Address extends IdEntity {

	//省份
	@Embedded
	private Division province;

	//市
	@Embedded
	private Division city;

	//区、县
	@Embedded
	private Division county;

	//坐标
	@Embedded
	private Coordinate coordinate;

	//街道
	private String street;

	private String streetNumber;

	//邮编号码
	private String zipcode;

	public Coordinate getCoordinate() {
		return coordinate;
	}

	public void setCoordinate(Coordinate coordinate) {
		this.coordinate = coordinate;
	}

	public Division getProvince() {
		return province;
	}

	public void setProvince(Division province) {
		this.province = province;
	}

	public Division getCity() {
		return city;
	}

	public void setCity(Division city) {
		this.city = city;
	}

	public Division getCounty() {
		return county;
	}

	public void setCounty(Division county) {
		this.county = county;
	}

	public String getStreetNumber() {
		return streetNumber;
	}

	public void setStreetNumber(String streetNumber) {
		this.streetNumber = streetNumber;
	}

	public String getStreet() {
		return street;
	}

	public void setStreet(String street) {
		this.street = street;
	}

	public String getZipcode() {
		return zipcode;
	}

	public void setZipcode(String zipcode) {
		this.zipcode = zipcode;
	}

	public Boolean isCoordinateEmpty() {
		if (coordinate == null) {
			return Boolean.TRUE;
		} else {
			return coordinate.isEmpty();
		}
	}

	public Boolean isDivisionEmpty(AdminLevel level) {
		switch (level) {
		case PROVINCE:
			return province == null ? Boolean.TRUE : province.isEmpty();
		case CITY:
			return city == null ? Boolean.TRUE : city.isEmpty();
		case COUNTY:
			return county == null ? Boolean.TRUE : county.isEmpty();
		default:
			return Boolean.TRUE;
		}
	}
}

File: Ceres-Framework/Ceres-Middleware-Location/src/main/resources/middleware/location/js/address.js
define([], function() {
	return {
		fromObject : function(obj) {
			obj = obj == undefined ? {} : obj;

			obj.isEmpty = function() {
				if (this.province == undefined || this.city == undefined || this.county == undefined || this.coordinate == undefined) {
					return false;
				}
				return true;
			}

			obj.toString = function() {
				var str = this.province == undefined ? "" : this.province.name == undefined ? "" : this.province.name;
				str = str + (this.city == undefined ? "" : this.city.name == undefined ? "" : this.city.name);
				str = str + (this.county == undefined ? "" : this.county.name == undefined ? "" : this.county.name);
				str = str + (this.street == undefined ? "" : this.street);
				str = str + (this.streetNumber == undefined ? "" : this.streetNumber);
				return str;
			}

			return obj;
		}
	}
});